tests/generate_p_files.sh

```diff
@@ -0,0 +1,21 @@
+#! /bin/sh
+#
+# Author:  guy maurel
+# May 2022
+#
+# generate p-files from the sources
+#
+# the test is only usefull for the developpers
+#
+LIST=`ls -1 ../src/*.cpp`
+LIST_H=`ls -1 ../src/*.h`
+P_FILES="../build/P-files"
```

Comment:
The paths are hard-coded. It would be more beneficial to get the paths from cmake during the configuration stage, so that the script can run in other folders too. For example I don't build in the ```build``` folder.

Reply:
yes, you are right